packages/shared/src/devServer.ts

```diff
@@ -14,13 +15,40 @@ import { DEFAULT_PORT, DEFAULT_DEV_HOST } from './constants';
 import { createAsyncHook } from './createHook';
 import type { Compiler } from '@rspack/core';
 
+export const normalizeUrl = (url: string) => url.replace(/([^:]\/)\/+/g, '$1');
```

Comment:
Can we add some unit tests for this method, RegExp is difficult to understand.

Reply:
done